redwood-tooling-codegen/src/test/kotlin/app/cash/redwood/tooling/codegen/TestingGenerationTest.kt

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (C) 2023 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package app.cash.redwood.tooling.codegen
+
+import app.cash.redwood.schema.Children
+import app.cash.redwood.schema.Default
+import app.cash.redwood.schema.Property
+import app.cash.redwood.schema.Schema
+import app.cash.redwood.schema.Widget
+import app.cash.redwood.tooling.schema.parseSchema
+import com.google.common.truth.Truth.assertThat
+import org.junit.Test
+
+class TestingGenerationTest {
+  @Schema(
+    [
+      BasicWidget::class,
+    ],
+  )
+  interface HappyPathSchema
+
+  @Widget(1)
+  data class BasicWidget(
+    @Property(1)
+    @Default("\"test\"")
+    val trait: String,
+    @Property(2)
+    @Default("{ error(\"test\") }")
+    val onEvent: () -> Unit,
+    @Children(1)
+    @Default("{}")
+    val block: () -> Unit,
+  )
+
+  @Test fun `tester happy path`() {
```

Comment:
I think we've almost entirely eliminated our tests which look like this. I would much prefer black-box tests, but I think your current design prohibits it with a bit _too_ much encapsulation.

I have an idea, but it's worth exploring after you merge as-is.

Reply:
Yeah, the test isn’t great 'cause it’s whitespace-sensitive. But it was also very convenient to have while writing the thing.

I suspect the most value it could offer going forward is if we introduced new cross-cutting features (say, changing modifiers), we’d get an example of how that changes the output.

Reply:
You can put types like the widget above into `test-schema/` and add a codegen module for its test code. That also lets us pull in the generated code into tests for the Redwood modules.

Your change of an implementation-detail type from `public` to `internal` just broke this test. I'm pretty confident that with a small tweak to the generated code we can drive it without the tester to validate it behaves as expected.

Reply:
Can do. Though I claim that visibility isn’t just an implementation detail, and having tests for it is nice.

Reply:
I have a branch to add API tracking of generated code: https://github.com/cashapp/redwood/compare/trunk...jw/metalava/2022-10-20. Presumably we'll also enable it for all the libraries, too. I haven't landed it because there's been too much churn, but we're nearly at a relatively stable state.

Reply:
ALSO - test schema caught lots of bugs. Good call.

Reply:
You don't have to change anything in this PR, though. We can always follow up.